featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -146,23 +152,79 @@ def test_init_es_with_relationships(pd_df):
     forward_dataframes = [name for name, _ in es.get_forward_dataframes('second_table')]
     assert forward_dataframes[0] == 'first_table'
 
+    relationship = es.relationships[0]
+    assert 'foreign_key' in relationship.child_column.ww.semantic_tags
+    assert 'index' in relationship.parent_column.ww.semantic_tags
 
-def test_add_secondary_time_index():
-    df = pd.DataFrame({
+
+@pytest.fixture
+def dates_df():
+    return pd.DataFrame({
         'backwards_order': [8, 7, 6, 5, 4, 3, 2, 1, 0],
         'dates_backwards': ['2020-09-09', '2020-09-08', '2020-09-07', '2020-09-06', '2020-09-05', '2020-09-04', '2020-09-03', '2020-09-02', '2020-09-01'],
         'random_order': [7, 6, 8, 0, 2, 4, 3, 1, 5],
         'repeating_dates': ['2020-08-01', '2019-08-01', '2020-08-01', '2012-08-01', '2019-08-01', '2019-08-01', '2019-08-01', '2013-08-01', '2019-08-01'],
         'special': [7, 8, 0, 1, 4, 2, 6, 3, 5],
         'special_dates': ['2020-08-01', '2019-08-01', '2020-08-01', '2012-08-01', '2019-08-01', '2019-08-01', '2019-08-01', '2013-08-01', '2019-08-01'],
     })
-    df.ww.init(index='backwards_order', time_index='dates_backwards')
+
+
+def test_add_secondary_time_index(dates_df):
+    dates_df.ww.init(index='backwards_order', time_index='dates_backwards')
     es = EntitySet('es')
-    es.add_dataframe('dates_table', df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+    es.add_dataframe('dates_table', dates_df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+
+    assert dates_df.ww.metadata['secondary_time_index'] == {'repeating_dates': ['random_order', 'special', 'repeating_dates']}
+
 
+def test_time_type_check_order(dates_df):
+    dates_df.ww.init(index='backwards_order', time_index='random_order')
+    es = EntitySet('es')
+
+    error = 'dates_table time index is numeric type which differs from other entityset time indexes'
+    with pytest.raises(TypeError, match=error):
+        # Because we set secondary time index before checking that time_index is valid, the time type will match the secondary time index
+        es.add_dataframe('dates_table', dates_df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
```

Comment:
In `EntitySet.add_dataframe`, we set the secondary time index before checking uniform time index for a set `time_index`. 

We still get an error if the time index type doesn't match the secondary time index type, but it seems a little odd to me that we set the time type based off of the secondary time index and not the time index. 

Is there any reason to not rearrange the order that we set the secondary time index?

Reply:
Seems like we should flip the order and set the time type from the time index and not the secondary time index. I can't think of a reason to keep the order as is.